xwiki-platform-core/xwiki-platform-skin/xwiki-platform-skin-api/src/main/java/org/xwiki/skin/Resource.java

```diff
@@ -57,6 +60,17 @@ public interface Resource<I extends InputSource>
      */
     I getInputSource() throws Exception;
 
+    /**
+     * @return the last modified time of that resource
+     * @throws Exception when failing to get the instance of the resource
```

Comment:
You are thinking about servlet resource based Resource implementations (because I was too lazy to try to make the difference between a failure caused by a type of URL for which it's expected to not be able to get the instance and a type for which the problem is different), but wiki entity based implements (in the case of a wiki skin for example) can throw an exception (when failing to access the document).

Reply:
Ok and you really prefer to throw the exception there, than to fallback on a null value?

Reply:
I prefer to make the distinction between:
* we cannot know the instant of this resource, but there is nothing wrong with it
* there is a real problem with the resource (and trying to access the content will most probably fail with a similar error), and we should be able to know the instant of this resource